.github/workflows/integration-tests.yml

```diff
@@ -18,18 +18,7 @@ jobs:
     steps:
 
       - name: Checkout repository
-        uses: actions/checkout@v2
```

Comment:
Maybe this might be useful too now? 

 > The go build cache should still be useful. Because we were probably [trashing](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy) all of our cache space on Github Actions with all the Docker stuff this wasn't effective. But if we get rid of this, it should be fine.

Reply:
just look at the caches generated, the fallbacks are not really working imho, so we are generating the same cache for each integration test. One run of the CI generates 8GB in cache.
I can take another look and see if stuff is properly being reused, but the steps of restoring and saving the caches are currently also taking very long